basicsr/archs/hifacegan_arch.py

```diff
@@ -0,0 +1,284 @@
+'''
+  HiFaceGAN_arch: Network definition of HiFaceGAN
+'''
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+# 20210519 [Lotayou]: Currently Pytorch official SyncBatchNorm only works under DDP training.
+# from basicsr.models.archs.sync_batchnorm import SynchronizedBatchNorm2d
+from basicsr.utils.registry import ARCH_REGISTRY
+from .hifacegan_util import BaseNetwork, LIPEncoder, SPADEResnetBlock, get_nonspade_norm_layer
+
+
+@ARCH_REGISTRY.register()
+class SPADEGenerator(BaseNetwork):
+    """Generator with SPADEResBlock"""
+
+    def __init__(self,
+                 num_in_ch=3,
+                 num_feat=64,
+                 use_vae=False,
+                 z_dim=256,
+                 num_upsampling_layers='normal',
+                 crop_size=512,
```

Comment:
the name of `num_upsampling_layers`  is misleading, I think just pass the number 5,6,7 is enough.

Reply:
It's all legacy code, case 6 & 7 is never used. Hardcode to 5 now.